src/main/main.ts

```diff
@@ -308,11 +324,29 @@ async function initializeApp() {
 }
 
 function initializeIpc() {
-  ipcMain.on("encounter different version", async (_event, data: Update) => {
-    if (data.extras) {
-      await update(data, updateOptions);
+  ipcMain.on(
+    "encounter different version",
+    async (_event, apv: Pick<AppProtocolVersionType, "version" | "extra">) => {
+      if (useUpdate) {
+        const decodedExtra = apv.extra && decodeApvExtra(apv.extra);
+
+        const simpleApv = {
+          version: apv.version,
+          extra: decodedExtra ? Object.fromEntries(decodedExtra) : {},
+        };
+
+        const context = await checkUpdateRequiredUsedPeersApv(
+          simpleApv,
+          standalone,
+          process.platform,
+          netenv,
+          baseURL,
+          getConfig("AppProtocolVersion")
+        );
+        await update(context, updateOptions);
+      }
```

Comment:
```suggestion
      if (!useUpdate) return;
      const extra = apv.extra ? Object.fromEntries(decodeApvExtra(apv.extra)) : {};

      const simpleApv = {
        version: apv.version,
        extra,
      };

      const context = await checkUpdateRequiredUsedPeersApv(
        simpleApv,
        standalone,
        process.platform,
        netenv,
        baseURL,
        getConfig("AppProtocolVersion")
      );
      await update(context, updateOptions);
```

Reply:
```typescript
 const extra = apv.extra ? Object.fromEntries(decodeApvExtra(apv.extra)) : {};
```
Unfortunately, decodeApvExtra return value is nullable

Reply:
rest code fixed!

Reply:
I see.